compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -583,6 +585,38 @@ generateSrcFromLf env = noLoc mod
             | otherwise
             = pure emptyBag
 
+    interfaceDecls :: [Gen (LHsDecl GhcPs)]
+    interfaceDecls = do
+        iface <- NM.toList $ LF.moduleInterfaces $ envMod env
+        [interfaceName] <- [LF.unTypeConName $ LF.intName iface]
+        let interfaceType = HsTyVar noExt NotPromoted $ mkRdrName interfaceName
+        meth <- NM.toList $ LF.intMethods iface
+        pure $ do
+            methodType <- convType env reexportedClasses $ LF.ifmType meth
+            let cls = HsTyVar noExt NotPromoted $ mkRdrName "HasMethod"
```

Comment:
I'm not satisfied with this since it's not explicitly `DA.Internal.Desugar.HasMethod`, but it seems to get the job done for now.

I saw a few functions at the bottom of this file which look useful (e.g. `mkGhcType`, `mkLfInternalType`) but those appear to be specifically for stable types/modules, which this is not.

Reply:
ah glad to hear that, I'll go with one of those then. Thanks! I guess we can come up with a different name later :)